shared/crypto/dkg.go

```diff
@@ -26,6 +26,20 @@ func MarshalDistKey(key *share.PriShare) ([]byte, error) {
 	return buf.Bytes(), nil
 }
 
+func UnmarshalDistKey(scheme ThresholdScheme, bytes []byte) (share.PriShare, error) {
+	I := binary.BigEndian.Uint64(bytes[0:8])
```

Comment:
This could cause a remote crash I think:
https://go.dev/play/p/PSO1zKw66p9

> panic: runtime error: slice bounds out of range [:8] with capacity 5

You might want to check the length first.

Reply:
updated

Reply:
added a test too for posterity